File: mecab.php
<?php
/*
 * Copyright (c) 2020 Nozomu Yamazaki
 * Released under the MIT license
 * https://opensource.org/licenses/mit-license.php
 * 
 * analyze text using mecab
 */

require 'vendor/autoload.php';

use Youaoi\MeCab\MeCab;

MeCab::setDefaults([
    // mecabのPATH
    'command' => posix_getpwuid(posix_geteuid())['dir'] . '/usr/local/bin/mecab',

    // 独自の辞書ディレクトリ
    'dictionaryDir' => posix_getpwuid(posix_geteuid())['dir'] . '/usr/local/lib/mecab/dic/ipadic/',

    // ユーザー辞書
    'dictionary' => posix_getpwuid(posix_geteuid())['dir'] . '/usr/local/lib/mecab/dic/user_dic/user_symbols.dic',
]);

/**
 * analyze the given string with mecab
 * @param string $text
 * @return array
 */
function analyze(string $text)
{
    $mecab = new MeCab();
    // change encoding from UTF-8 to EUC-JP
    $array = $mecab->analysis(mb_convert_encoding($text, 'EUC-JP', 'UTF-8'));

    $result = array();
    foreach ($array as $word_num => $a) {
        $a = (array) $a;
        // change encoding from EUC-JP to UTF-8
        mb_convert_variables('UTF-8', 'EUC-JP', $a);
        $result_word = array();
        foreach ($a as $key => $str) {
            if (strpos($key, "str")) {
                $result_word['str'] = $str;
            } else if (strpos($key, "text")) {
                $result_word['text'] = $str;
            } else if (strpos($key, "speechInfo")) {
                foreach ($str as $num => $s) {
                    if ($num == 0) {
                        $result_word['speechInfo1'] = $s;
                    } elseif ($num == 1) {
                        $result_word['speechInfo2'] = $s;
                    } elseif ($num == 2) {
                        $result_word['speechInfo3'] = $s;
                    }
                }
            } else if (strpos($key, "speech")) {
                $result_word['speech'] = $str;
            } else if (strpos($key, "conjugateType")) {
                $result_word['conjugateType'] = $str;
            } else if (strpos($key, "conjugate")) {
                $result_word['conjugate'] = $str;
            } else if (strpos($key, "original")) {
                $result_word['original'] = $str;
            } else if (strpos($key, "reading")) {
                $result_word['reading'] = $str;
            } else if (strpos($key, "pronunciation")) {
                $result_word['pronounciation'] = $str;
            }
        }
        $speech = $result_word['speech'];
        if ($speech == "助詞" || $speech == "記号" || $speech == "助動詞" || $speech == "その他,間投" || $speech == "フィラー" || $speech == "連体詞") {
            continue;
        }
        $result[$word_num] = $result_word;
    }
    return $result;
}

File: crawler/mecab.py
import pprint
import MeCab


def mecab(text):
    user_dictionary = "/usr/local/lib/mecab/dic/user_dic/user_dic.dic"
    dictionary_dir = "/usr/local/lib/mecab/dic/ipadic/"

    mecab = MeCab.Tagger("--userdic=" + user_dictionary + " --dicdir=" + dictionary_dir)

    node = mecab.parseToNode(text)

    result = []
    while node:
        result_word = {}
        pos = node.feature.split(",")
        if pos[0] != "BOS/EOS":
            result_word['str'] = node.surface + '\t' + node.feature
            result_word['text'] = node.surface
            for i in range(len(pos)):
                if pos[i] == "*":
                    pos[i] = ""
            result_word['speech'] = pos[0]
            result_word['speechInfo1'] = pos[1]
            result_word['speechInfo2'] = pos[2]
            result_word['speechInfo3'] = pos[3]
            result_word['conjugate'] = pos[4]
            result_word['conjugateType'] = pos[5]
            result_word['original'] = pos[6]
            if len(pos) == 9:
                result_word['reading'] = pos[7]
                result_word['pronaounciation'] = pos[8]
            else:
                result_word['reading'] = ""
                result_word['pronaounciation'] = ""
            speech = result_word['speech']
            if speech == "助詞" or speech == "記号" or speech == "助動詞" or speech == "その他,間投" or speech == "フィラー" or speech == "連体詞":
                pass
            else:
                result.append(result_word)
        node = node.next
    return result


if __name__ == '__main__':
    word = "すもももももももものうち"

    result = mecab(word)
    pprint.pprint(result)
